minesweeper.py

```
@@ -0,0 +1,196 @@
+                mines_qty += 1
+
+        return mines_qty
+
+    def click_square(self, row, col):
+        """Updates clicked to True on game board for specified row and col."""
+
+        if self.board[row][col].clicked:
+            print self.board[row][col]
+            print 'You already clicked that square. Choose another.'
+        else:
+            self.board[row][col].clicked = True
+
+    def flag_square(self, row, col):
+        """Flags a square with a question mark."""
+
+        self.board[row][col].flagged = True
+
+    def is_mine(self, row, col):
+        """Returns True if specified row and col is a mine."""
+
+        return self.board[row][col]._mine
+
+    def display_board(self):
+        """Visually represent state of board."""
+
+        print ' ',
+        for n in range(self.width):
+            print ' {} '.format(n),
+        print ''
+        for i, row in enumerate(self.board):
+            print i,
+            for col in self.board[row]:
+                if self.board[row][col].clicked:
+                    if self.board[row][col]._mine:
+                        print '[*]',
+                    else:
+                        mines = self._count_touching_mines(row, col)
+                        print ' {} '.format(mines),
+                elif self.board[row][col].flagged:
+                    print '[?]',
+                else:
+                    print '[ ]',
+            print '\n'
+
+
+class Minesweeper(object):
+
+    def __init__(self):
+        self.keep_playing = True
+        self.current_game = Board()
+        self._click_qty = 0
+        self.start_game()
+
+    def start_game(self):
+        """Initiates minesweeper game."""
+
+        print 'Welcome to Minesweeper!'
+        response = raw_input('Current board is 10x10 with 20 mines. Press enter to begin, or type CUSTOM to modify board. > ')
+
+        if response.lower() == 'custom':
+            rows = raw_input('How many rows? > ')
```

Seems like the cast to int should occur here, rather than end the end so it fails fast on bad user input.